### PR TITLE
Move GlanceAPI deployment to Statefulset

### PR DESCRIPTION
**Pull request by fmount**
This patch improves the way the glance statefulset uses PVCs:
   
- `ensurePVC` is not required anymore and the logic has been moved away from controller to the `statefulset` creation function
- `storageClass` and `storageRequest` are now fields of the `GlanceAPITemplate` 
- `getVolumes` don't require `pvc(s)` anymore
- `DBSync` only mounts what it needs (https://github.com/openstack-k8s-operators/glance-operator/pull/351)
- `GlanceAPI` `Pvc` is moved from `RWX` to `RWO`

### CODE REVIEW

**Comment by fmount** [comment]
@mdbooth and I tested it in a live (baremetal) environement and it works as expected. 
/cc @fultonj

File: api/v1beta1/glance_types.go [hunk 1]
```diff
@@ -167,12 +167,6 @@ type GlanceStatus struct {
 
 	// Glance Database Hostname
 	DatabaseHostname string `json:"databaseHostname,omitempty"`
-
-	// ReadyCount of internal and admin Glance API instance
-	GlanceAPIInternalReadyCount int32 `json:"glanceAPIInternalReadyCount,omitempty"`
```
Comment (fmount) [inline api/v1beta1/glance_types.go:172]: @abays this is not required anymore w/ statefulset

File: config/samples/import_plugins/image_conversion/image_conversion_pvc.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ metadata:
   name: image-import-staging-workspace
 spec:
   accessModes:
-  - ReadWriteMany
+  - ReadWriteOnce
```
Comment (fmount) [inline config/samples/import_plugins/image_conversion/image_conversion_pvc.yaml:7]: each glance pod is going to have stable access to is own pvc

```
glance-0 -> pvc-0
glance-1 -> pvc-1
glance-2 -> pvc-2
```

File: controllers/glance_controller.go [hunk 3]
```diff
@@ -284,15 +283,6 @@ func (r *GlanceReconciler) reconcileInit(
 ) (ctrl.Result, error) {
 	r.Log.Info(fmt.Sprintf("Reconciling Service '%s' init", instance.Name))
 
-	// Define the PVCs objects required by Glance
-	ctrlResult, err := r.ensurePVC(ctx, helper, instance, serviceLabels)
```
Comment (fmount) [inline controllers/glance_controller.go:288]: moving to `statefulset` in the first place allow us to remove the PV creation logic from the `glance-controller`

File: controllers/glance_controller.go [hunk 4]
```diff
@@ -699,6 +687,10 @@ func (r *GlanceReconciler) apiDeploymentCreateOrUpdate(ctx context.Context, inst
 		apiSpec.GlanceAPITemplate.NodeSelector = instance.Spec.NodeSelector
 	}
 
+	// Inherit the values required for PVC creation from the top-level CR
+	apiSpec.GlanceAPITemplate.StorageRequest = instance.Spec.StorageRequest
```
Comment (fmount) [inline controllers/glance_controller.go:691]: All the instances needs to be tied to the same `StorageClass` and the same `StorageRequest`, hence this propagates the two fields to the `statefulset` that is supposed to create them via `VolumeClaimTemplates`

File: controllers/glanceapi_controller.go [hunk 5]
```diff
@@ -44,7 +44,7 @@ import (
 	keystonev1 "github.com/openstack-k8s-operators/keystone-operator/api/v1beta1"
 	"github.com/openstack-k8s-operators/lib-common/modules/common"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
-	"github.com/openstack-k8s-operators/lib-common/modules/common/deployment"
+	//"github.com/openstack-k8s-operators/lib-common/modules/common/deployment"
```
Comment (fmount) [inline controllers/glanceapi_controller.go:47]: I have to remove this

Comment (fmount) [reply controllers/glanceapi_controller.go]: Done

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**